src/util/helpers.js

```diff
@@ -99,21 +99,63 @@ export function addOnceEventListener (el, event, cb) {
   el.addEventListener(event, once, false)
 }
 
-export function getObjectValueByPath (obj, path) {
+export function getNestedValue (obj, path, fallback) {
+  const last = path.length - 1
+
+  if (last < 0) return obj === undefined ? fallback : obj
+
+  for (let i = 0; i < last; i++) {
+    if (obj === null || obj === undefined) {
+      return fallback
+    }
+    obj = obj[path[i]]
+  }
+
+  if (obj === null || obj === undefined) return fallback
+
+  return obj[path[last]] === undefined ? fallback : obj[path[last]]
+}
+
+export function looseEqual (a, b) {
+  if (a === b) return true
+
+  if (a !== Object(a) || b !== Object(b)) {
+    // If the values aren't objects, they were already checked for equality
+    return false
+  }
+
+  const props = Object.keys(a)
+
+  if (props.length !== Object.keys(b).length) {
+    // Different number of props, don't bother to check
+    return false
+  }
+
+  return props.every(p => looseEqual(a[p], b[p]))
+}
+
+export function getObjectValueByPath (obj, path, fallback) {
   // credit: http://stackoverflow.com/questions/6491463/accessing-nested-javascript-objects-with-string-key#comment55278413_6491621
-  if (!path || path.constructor !== String) return
+  if (!path || path.constructor !== String) return fallback
```

Comment:
Any reason you're using `constructor` rather than `typeof`?

Reply:
class X extends String {}
Anyway, it was like that before my commits. So I really don't know the reason.

Reply:
Ah right, didn't notice that in the diff.